lang/cs/Org.Apache.REEF.IMRU.Examples/PipelinedBroadcastReduce/BroadcastSenderReduceReceiverUpdateFunction.cs

```diff
@@ -84,12 +87,52 @@ internal sealed class BroadcastSenderReduceReceiverUpdateFunction : IUpdateFunct
         /// <returns>Map input</returns>
         UpdateResult<int[], int[]> IUpdateFunction<int[], int[], int[]>.Initialize()
         {
-            for (int i = 0; i < _dim; i++)
+            if (_taskState != null && _taskState.Result != null && (_taskState.Result.HasMapInput || _taskState.Result.HasResult))
```

Comment:
It won't in this case as we have injected it. But no harm to check . The idea is for any reason we cannot get intermediate result, we will start from fresh initialization.